packages/components/src/components/teaser/teaser.ts

```diff
@@ -0,0 +1,163 @@
+import { HasSlotController } from '../../internal/slot';
+import cx from 'classix';
+import SolidElement from '../../internal/solid-element';
+
+/**
+ * @summary Teasers group information into flexible containers so users can browse a collection of related items and actions.
+ * @documentation https://solid.union-investment.com/[storybook-link]/teaser
+ *
+ * @status stable
+ * @since 1.2
+ * *
+ * @slot - An optional main content slot.
+ * @slot media - An optional media slot.
+ * @slot meta - An optional meta slot.
+ * @slot headline - headline slot.
+ *
+ * @csspart base - The component's base wrapper.
+ * @csspart media - The container that wraps the media.
+ * @csspart meta -  The container that wraps the meta.
+ * @csspart headline - The container that wraps the headline.
+ * @csspart main - The container that wraps the main content.
+ *
+ * @cssproperty --distribution-media - The distribution ratio of the media.
+ * @cssproperty --distribution-content - The distribution ratio of the content.
+ */
+
+@customElement('sd-teaser')
+export default class SdTeaser extends SolidElement {
+  @property({ reflect: true })
+  variant: 'white' | 'white border-neutral-300' | 'neutral-100' | 'primary' | 'primary-100' = 'white';
+
+  /** The teaser's orientation */
+  @property({ reflect: true })
+  orientation: 'responsive' | 'vertical' | 'horizontal' = 'responsive';
+
+  /** The teaser's padding */
+  @property({ type: Boolean, reflect: true }) inset = false;
+
+  @query('[part="base"]') teaser: HTMLElement;
+
+  @state() private currentOrientation: 'vertical' | 'horizontal';
+
+  private readonly hasSlotController = new HasSlotController(this, '[default]', 'media', 'meta', 'headline');
+
+  private resizeObserver: ResizeObserver;
+
+  connectedCallback() {
+    super.connectedCallback();
+    if (window.ResizeObserver) {
+      this.resizeObserver = new ResizeObserver(() => this.updateOrientation());
+      this.resizeObserver.observe(this);
+    }
+  }
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    if (this.resizeObserver) {
+      this.resizeObserver.disconnect();
+    }
+  }
+
+  updateOrientation() {
+    if (this.orientation === 'responsive') {
+      this.currentOrientation = this.offsetWidth > 448 ? 'horizontal' : 'vertical';
+      this.requestUpdate();
+    } else {
+      this.currentOrientation = this.orientation;
+    }
+  }
+
+  render() {
+    if (this.variant === 'white border-neutral-300') {
+      this.inset = true;
+    }
+
+    const slots = {
+      default: this.hasSlotController.test('[default]'),
+      media: this.hasSlotController.test('media'),
+      meta: this.hasSlotController.test('meta'),
+      headline: this.hasSlotController.test('headline')
+    };
+
+    return html`
+      <div
+        class=${cx(
+          'flex',
+          {
+            white: 'bg-white',
+            'white border-neutral-300': 'bg-white border border-neutral-300',
```

Comment:
That's a good idea to split this with a space! So the first one is always the background and then comes the border.

When splitting it with a space, we could even name name it `white neutral-300` like design proposed – what do you prefer?

Reply:
Hmm, was it proposed in design? This is the name I got from figma, I feel like it makes sense because it makes it clear that this variant has a border unlike the others, but I guess both work!

Reply:
If it makes it more clear, let's leave it like it is. :)